### PR TITLE
feat(QF-4655): redesign CourseDetails completed state with inline pill and repeat learning plan button 

### PR DESCRIPTION
**Pull request by yousefdergham**
## Summary                                                                                                                            
                                                                                                                                        
  Redesigns the CourseDetails component for completed courses:                                                                          
  - Moves the "Completed" pill inline with the title text so it flows naturally with wrapping                                           
  - Introduces a new "Repeat Learning Plan" button that navigates to the first lesson                                                   
  - Separates title+pill and action buttons into two rows on desktop                                                                    
  - update lp mobile layout: image first, then title+pill, then buttons                                                                  
  - Removes the old `CompletedStatus` component (dead code after refactor)                                                              

  Closes: [QF-4655](https://quranfoundation.atlassian.net/browse/QF-4655)

  ## Type of Change

  - [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
  - [x] ✨ New feature (non-breaking change which adds functionality)
  - [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [ ] 📝 Documentation update
  - [x] ♻️ Refactoring (no functional changes)

  ## Scope Confirmation

  - [x] This PR addresses **one** feature/fix only
  - [x] If multiple changes were needed, they are split into separate PRs

  ## Rollback Safety

  - [x] Can be safely reverted without data issues or migrations
  - [ ] Rollback requires special steps (describe below):

  ## Test Plan

  - [ ] Unit tests added/updated
  - [ ] Integration tests added/updated
  - [x] Manual testing performed

  **Testing steps:**

  1. Navigate to a **completed** learning plan — verify title + COMPLETED pill are inline, "Repeat Learning Plan" and "Add Feedback"
  buttons appear right-aligned on a second row below the title
  2. Resize to mobile (<768px) — verify image appears first, title (24px) + pill below, buttons (flex-end) below that with 20px gap to
  image
  3. Click "Repeat Learning Plan" — verify it navigates to the first lesson of the course
  4. Navigate to a **non-enrolled** learning plan — verify "Start Here" button is to the right of the title on desktop, and on a new row (right-aligned) on mobile
  5. Navigate to an **enrolled but not completed** learning plan — verify "Continue Learning" button works as before
  6. Switch to RTL language (Arabic) — verify layout mirrors correctly
  7. Switch to dark/sepia theme — verify Repeat button background adapts

  ### Edge Cases Verified

  - [ ] ⏳ Loading state handled
  - [ ] ❌ Error state handled
  - [ ] 📭 Empty state handled
  - [x] 👤 Logged-in vs guest behavior (if applicable)

  ## Pre-Review Checklist

  ### Code Quality

  - [x] I have performed a **self-review** of my code (file by file)
  - [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
  - [x] No `any` types used (or justified if unavoidable)
  - [x] No unused code, imports, or dead code included
  - [x] Complex logic has inline comments explaining "why"
  - [x] Functions are under 30 lines and follow single responsibility

  ### Testing & Validation

  - [ ] All tests pass locally (`yarn test`)
  - [ ] Linting passes (`yarn lint`)
  - [ ] Build succeeds (`yarn build`)
  - [x] Edge cases and error scenarios are handled

  ### Documentation

  - [x] Code is self-documenting with clear naming
  - [ ] README updated (if adding features or setup changes)
  - [ ] Inline comments added for complex logic

  ### Localization (if UI changes)

  - [x] All user-facing text uses `next-translate`
  - [x] Only English locale files modified (Lokalise handles others)
  - [x] RTL layout verified

  ### Accessibility (if UI changes)

  - [x] Semantic HTML elements used
  - [ ] ARIA attributes added where needed
  - [x] Keyboard navigation works

  ## AI Assistance Disclosure

  - [ ] AI tools were NOT used for this PR
  - [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code


[QF-4655]: https://quranfoundation.atlassian.net/browse/QF-4655?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Course/CourseDetails/CourseDetails.module.scss [hunk 1]
```diff
@@ -5,19 +5,69 @@
...
+    font-size: var(--font-size-large-px);
+    align-self: flex-start;
+  }
+}
+
+.completedPill {
+  vertical-align: middle;
+  margin-inline-start: var(--spacing-small-px);
+  font-size: var(--font-size-small-px);
+  color: var(--color-text-white);
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/Course/CourseDetails/CourseDetails.module.scss:55]: it's design rquirment
